tests/notebooks/e2e-wine/e2e-wine-kfp-mlflow-seldon.ipynb

```diff
@@ -0,0 +1,1233 @@
+    "execution": {
+     "iopub.execute_input": "2023-08-09T14:52:51.561354Z",
+     "iopub.status.busy": "2023-08-09T14:52:51.560055Z",
+     "iopub.status.idle": "2023-08-09T14:52:51.632436Z",
+     "shell.execute_reply": "2023-08-09T14:52:51.631115Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "run = mlflow.last_active_run()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "id": "50fdc7d2-1e4d-4a15-8731-01057aee72b2",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2023-08-09T14:52:51.637551Z",
+     "iopub.status.busy": "2023-08-09T14:52:51.637211Z",
+     "iopub.status.idle": "2023-08-09T14:52:51.646533Z",
+     "shell.execute_reply": "2023-08-09T14:52:51.645475Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "@retry(\n",
+    "    wait=wait_exponential(multiplier=2, min=1, max=10),\n",
+    "    stop=stop_after_attempt(30),\n",
+    "    reraise=True,\n",
+    ")\n",
+    "def assert_mlflow_run_finished(client, run_id):\n",
+    "    \"\"\"Wait for the run to complete successfully.\"\"\"\n",
+    "    status = client.get_run(run_id).info.status\n",
+    "    assert status == \"FINISHED\", f\"MLFlow run in {status} state.\"\n",
+    "\n",
+    "\n",
+    "def assert_mlflow_model_exists(client, run_id):\n",
+    "    \"\"\"Assert Model exists.\"\"\"\n",
+    "    model = client.sklearn.load_model(f\"runs:/{run_id}/model\")\n",
+    "    assert isinstance(model, ElasticNet), f\"Model {model} is not of type ElasticNet!\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "id": "836dd8a4-3caa-4438-9e1f-3d9e7307d668",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2023-08-09T14:52:51.650269Z",
+     "iopub.status.busy": "2023-08-09T14:52:51.649754Z",
+     "iopub.status.idle": "2023-08-09T14:52:51.872039Z",
+     "shell.execute_reply": "2023-08-09T14:52:51.870234Z"
+    },
+    "tags": [
+     "raises-exception"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "assert run.data.tags.get(\"author\") == \"kf-testing\"\n",
+    "assert_mlflow_run_finished(mlflow, run.info.run_id)\n",
+    "assert_mlflow_model_exists(mlflow, run.info.run_id)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "id": "a2606489-1dc6-47c1-adc0-d8e706cfbc8c",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2023-08-09T14:52:51.876437Z",
+     "iopub.status.busy": "2023-08-09T14:52:51.876126Z",
+     "iopub.status.idle": "2023-08-09T14:52:51.947490Z",
+     "shell.execute_reply": "2023-08-09T14:52:51.937517Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# workflow component\n",
+    "training_op = kfp.components.create_component_from_func(\n",
+    "    func=training,\n",
+    "    output_component_file=\"train-component.yaml\",\n",
+    "    base_image=\"python:3.8.10\",\n",
+    "    packages_to_install=[\"boto3\", \"mlflow==2.1.1\", \"numpy\", \"pandas\", \"pyarrow\", \"scikit-learn<1.2\"],\n",
```

Comment:
I know it is not very common to do this in notebooks, but I wonder if we should keep these package versions as configurable as possible so if charmed MLFlow suddenly needs different versions of the `mlflow` python package, for instance, we can easily change that, so this file is easy to maintain and easy to keep using in the future as our charms change.

Reply:
Yeah I know what you mean, because I was struggling to keep this pins in sync across the notebook. My problem is that notebook dependencies will be moved to a `requirements.txt` eventually, so I'm not sure what I can do about the component dependencies. This is a special situation anyway, given that almost all components are intended to be run both locally and as part of the Pipeline.

Reply:
I guess what we could do to prevent incompatibilities is to add a section at the top of this notebook that says which versions of our charms work (or should work) with this.

Reply:
Added a commit to address this!